pkg/services/notification/resource_control/address.go

```diff
@@ -81,12 +85,40 @@ func ModifyAddress(ctx context.Context, req *pb.ModifyAddressRequest) (string, e
 
 	attributes[models.AddrColStatusTime] = time.Now()
 
-	if err := dbutil.GetChain(globalcfg.GetInstance().GetDB().Table(models.TableAddress)).
+	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableAddress)).
 		Where(models.AddrColId+" = ?", addressId).
 		Updates(attributes).Error; err != nil {
-		logger.Errorf(ctx, "Update Address [%s] failed: %+v", addressId, err)
+		logger.Errorf(ctx, "Update address [%s] failed, [%+v]", addressId, err)
 		return "", err
 	}
 
 	return addressId, nil
 }
+
+func UpdateAddressAddrLsIdByIds(ctx context.Context, addrListId string, addrIds []string) ([]string, error) {
+	attributes := make(map[string]interface{})
+	attributes[models.AddrColAddrListId] = addrListId
+	attributes[models.AddrColStatusTime] = time.Now()
+
+	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableAddress)).
+		Where(models.AddrColId+" in (?)", addrIds).
+		Updates(attributes).Error; err != nil {
+		logger.Errorf(ctx, "Update address [%s] failed, [%+v]", addrIds, err)
+		return nil, err
+	}
+
+	return addrIds, nil
+}
+
+func DeleteAddresses(ctx context.Context, addressIds []string) ([]string, error) {
+	db := global.GetInstance().GetDB()
+	tx := db.Begin()
+	tx.Delete(models.Address{}, models.AddrColId+" in (?)", addressIds)
```

Comment:
Should not real delete, just update status to deleted.

Reply:
updated.